src/app/view/perspectives/search/documents/search-documents.component.ts

```diff
@@ -99,54 +110,40 @@ export class SearchDocumentsComponent implements OnInit, OnDestroy {
     return this.size === SizeType.XL;
   }
 
-  private initDocuments(documents: Document[]) {
-    const codes = new Set<string>();
-    this.documents = [];
-    for (let document of documents) {
-      delete document.data['_id'];
-      this.documents.push({document: document, opened: false});
-      codes.add(document.collectionCode);
-    }
-    codes.forEach(code => this.fetchCollection(code));
-  }
-
-  private fetchCollection(code: string) {
-    this.collectionService.getCollection(code)
-      .subscribe(collection => this.initCollectionsInDocuments(collection));
-  }
-
-  private initCollectionsInDocuments(collection: Collection) {
-    for (let document of this.documents) {
-      if (document.document.collectionCode === collection.code) {
-        document.collectionName = collection.name;
-        document.collectionIcon = collection.icon;
-        document.collectionColor = collection.color;
-      }
-    }
-  }
-
-  public createDefaultAttributeHtml(document: SearchDocument): string {
-    const data = document.document.data;
+  public createDefaultAttributeHtml(document: DocumentModel): string {
+    const data = document.data;
     return this.valueHtml(Object.values(data)[0]);
   }
 
-  public toggleDocument(document: SearchDocument) {
+  public toggleDocument(document: DocumentModel) {
     document.opened = !document.opened;
```

Comment:
You are breaking the immutability here. All changes of the store objects need to be done in the reducer only! So rather than doing this, you need to define an action which will mark the given document as opened.

However, I'm not sure if it is better to mix this in the documents store through `opened` field as you suggested or to create a new search store where there will be an array of expanded document IDs. The latter approach will have one advantage, it will allow to manage the expanded documents at one place and decide when you want to clear them. For example I think the expected behavior would be to keep them explanded when you change the search tabs but collapse them when you make a new search. What do you think?